### ISSUE TITLE
Unread messages icon missing from taskbar following 4.7.0 upgrade

### ISSUE THREAD

**Issue by bbodenmiller** [mattermost/desktop#1648]
I confirm (by marking "x" in the [ ] below: [x]):

- [ ] This is not a troubleshooting question. [Troubleshooting questions go here: http://www.mattermost.org/troubleshoot/](http://www.mattermost.org/troubleshoot/).
- [ ] This doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/platform/issues).
- [ ] I have read [contributing guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md).

---

**Summary**

Red unread messages icon missing from taskbar following 4.7.0 upgrade

**Environment**
- Operating System: Windows 10
- Mattermost Desktop App version: 4.7.0 
- Mattermost Server version: 5.33.3 

**Steps to reproduce**

1. Have desktop 4.6.1 intalled
2. Upgrade to 4.7.0
3. Verify option is enabled in settings

**Expected behavior**

Red unread messages icon shows in taskbar

**Observed behavior**

No red unread messages icon in taskbar. The icon for direct mentions/replies (red icon with number) appears to work fine.

**Possible fixes**

**Comment by bbodenmiller** [comment]
Exiting Mattermost & reopening did not seem to fix issue. Nor did exiting, shutting down `explorer.exe`, reopening `explorer.exe`, then reopening Mattermost. Oddly however selecting log out in app, exiting app, reopening, then relogging in seems to have brought back unread icon.

**Comment by bbodenmiller** [comment]
Possibly related: I'm getting the taskbar icon flashing on new message arrival as well despite having that option disabled in settings.

**Comment by Willyfrog** [comment]
hello @bbodenmiller and thanks for reporting, did you use the same type of installer in both instances? if you go to the apps configuration there is only one mattermost install, right?

**Comment by bbodenmiller** [comment]
@Willyfrog correct. I used `mattermost-desktop-setup-4.6.1-win.exe` and `mattermost-desktop-setup-4.7.0-win.exe`.

Only one Mattermost installed:

![image](https://user-images.githubusercontent.com/1192780/124641905-0d775980-de44-11eb-90b9-1e16d46e569e.png)

**Comment by Willyfrog** [comment]
yes, I can replicate with the same exes, I'll fill a [ticket to investigate further](https://mattermost.atlassian.net/browse/MM-36859)

**Comment by Willyfrog** [comment]
restarting windows seems to fix it, maybe the file was locked by the old version until restart?

**Comment by bbodenmiller** [comment]
@Willyfrog restarting didn't fix it for me. Earlier selecting log out in app, exiting app, reopening, then relogging brought back unread icon. Now that's not even working.

**Comment by Willyfrog** [comment]
if you go to the config file, which value does it have? does it change if you change it within the settings?

**Comment by bbodenmiller** [comment]
@Willyfrog values:

```json
{
  "teams": [
    {
      "name": "xxxx",
      "url": "xxxx",
      "order": 0
    }
  ],
  "showTrayIcon": true,
  "trayIconTheme": "light",
  "version": 2,
  "minimizeToTray": true,
  "notifications": {
    "flashWindow": 0,
    "bounceIcon": false,
    "bounceIconType": "informational"
  },
  "showUnreadBadge": true,
  "useSpellChecker": true,
  "spellCheckerLocale": "en-US",
  "enableHardwareAcceleration": true,
  "autostart": true,
  "darkMode": false
}
```

Changing settings in UI changes things in config file as expected. e.g. `"flashWindow": 2,`, `"showUnreadBadge": false,`

**Comment by Willyfrog** [comment]
can you go to the folder where the config is stored and remove everything but the json files? it might be something that is cached from a previous version.

the path to that folder usually is `c:\Users\<user>\AppData\Roaming\Mattermost`

**Comment by bbodenmiller** [comment]
@Willyfrog this may have worked for unread icon but it's not practical for us to tell all users to do this. Also it didn't fix https://github.com/mattermost/desktop/issues/1649.

**Comment by Willyfrog** [comment]
so far, this is the only report I have, hence the manual fix. If there are many other reporting the same problem then a fix would be needed.

**Comment by incx444** [comment]
Had the same thing with 4.7.0 desktop for several weeks, upgraded from 4.6.2. Taskbar would not reset the red numberless unread messages bubble unless I switched to another channel or person in the UI. In addition, after installing server 5.37 the bubble started appearing with no visible new traffic on the server. Went back to 4.6.2, all is working again.

**Comment by Willyfrog** [comment]
> Had the same thing with 4.7.0 desktop for several weeks, upgraded from 4.6.2. Taskbar would not reset the red numberless unread messages bubble unless I switched to another channel or person in the UI. In addition, after installing server 5.37 the bubble started appearing with no visible new traffic on the server. Went back to 4.6.2, all is working again.

hello and thanks for the report, this is a different bug that will be solved in 4.7.1

**Comment by amyblais** [comment]
Opened a ticket for the original report https://mattermost.atlassian.net/browse/MM-37807.

**Comment by LcHicks** [comment]
@Willyfrog, I am having the same issue as @bbodenmiller. For your stop-gap fix suggestion, do you mean to delete _everything_ in that folder except the `.json` files? Including all the folders? I have 129 items in that folder.

![image](https://user-images.githubusercontent.com/76702572/133819154-97e18142-7ef7-47e6-b5b2-284ca9eba326.png)

**Comment by Willyfrog** [comment]
yes, I'd suggest to first quit the app completely, and then remove anything that it is not a `.json` file. All the others will be generated on start, but the json ones are your config.

**Comment by LcHicks** [comment]
@Willyfrog I exited the mattermost app completely (from the system tray) and deleted all the non-`.json` files and folders. I can confirm I am still seeing this issue. There is a notification bubble on a non-selected team yet the taskbar alert icon is missing.

![image](https://user-images.githubusercontent.com/76702572/133859875-ade6bf8f-3a76-4427-a1d5-94fd9e5b72ed.png)

**Comment by Willyfrog** [comment]
@LcHicks which version of mattermost are you running?

**Comment by LcHicks** [comment]
The desktop client is 4.7.2, the server is 5.33.3

**Comment by LcHicks** [comment]
This problem is still present in desktop client version 5.0.1 and server versions as follows:
Mattermost Version: 5.39.2
Database Schema Version: 5.38.2

**Comment by LcHicks** [comment]
## Update

@devinbinnie I have now found that my fix above only partway fixed the issue.

My company has many different Mattermost teams. I am a part of 14 different teams.
I have found the the fix I did above enables the notification pip to show up only when the notification is for the team I have selected to view. If any other teams have notifications, the pip doesn't show up. It doesn't matter what team was last selected, the behavior is the same: notification pip is only showing notifications for the currently selected team.

**Comment by devinbinnie** [comment]
> 

That was fixed in this PR: https://github.com/mattermost/desktop/pull/2020
The fix is available in the current nightly build available here: https://community-daily.mattermost.com/core/channels/desktop-builds

It will be released as part of v5.1 in May.

**Comment by LcHicks** [comment]
@devinbinnie, another update from today:
Each day I shutdown my work computer at the end of the day. When I booted it up today and started using Mattermost, the problem I first had re-appeared (red notification pip on taskbar icon doesn't show up unless the Mattermost program is the "focused" program).

I then did my same fix as before and the notification pip was working again today.

Therefore, there's something happening upon computer shutdown/start-up that is causing the issue to come back.  
I have tested exiting out of Mattermost and then re-opening and the notification pip was still working. So the reversion back to failed functionality is happening upon shutdown/power-up and not on Mattermost exit/re-opening.

**Comment by devinbinnie** [comment]
Is there something on your work computer that might be verifying files perhaps? Or stopping the app from writing to the tray icon? Since it's a work computer I wouldn't be surprised if some security applications may be restricting access.

**Comment by devinbinnie** [comment]
Closing as inactive, feel free to re-open if this is still an issue on v5.3
